desktop-app/app/components/ScreenshotManager/Capture.js

```diff
@@ -0,0 +1,268 @@
+// @flow
+
+import React from 'react';
+import type {Data} from './types';
+import WebViewUtils from './screenShotUtils';
+import FileSystemUtils from '../../utils/file-system';
+import {userPreferenceSettings} from '../../settings/userPreferenceSettings';
+import {toast} from 'react-toastify';
+import NotificationMessage from '../NotificationMessage';
+import {
+  SCREENSHOT_IN_PROGRESS,
+  TOGGLE_EVENT_MIRRORING_ALL_DEVICES,
+} from '../../constants/pubsubEvents';
+import pubsub from 'pubsub.js';
+import {
+  convertNativeImageToJPEG,
+  convertURIImageToBuffer,
+} from '../../utils/imageUtils';
+import {_delay} from './functionUtils';
+
+function sortWebViewsByDeviceIds(
+  webViews: NodeList<WebviewElement>,
+  devices: Array<{id: number}>,
+  selectedDevices
+): Array<any> {
+  const hashTable = new Map<number, number>();
+  const sorted = new Array(devices.length).fill(null);
+
+  for (let i = 0; i < devices.length; i += 1) {
+    hashTable.set(devices[i].id, i);
+  }
+
+  for (let i = 0; i < webViews.length; i += 1) {
+    const id = webViews[i].id;
+    if (hashTable.has(id)) {
+      const index = hashTable.get(id);
+      sorted[index] = webViews[i];
+    }
+  }
+  const selected = [];
+  for (let i = 0; i < sorted.length; i += 1) {
+    if (selectedDevices[i].selected) {
+      selected.push(sorted[i]);
+    }
+  }
+  return selected;
+}
+
+export default async function Capture(data: Data) {
+  const {
+    now,
+    devices,
+    selectedDevices: {deviceChecks, isMergeImages},
+  } = data;
+
+  // get All web views
+  const webViews = document.querySelectorAll('webview');
+
+  // sort web views based on devices order
+  const sortedWebViews = sortWebViewsByDeviceIds(
+    webViews,
+    devices,
+    deviceChecks
+  );
+
+  // get user preference path
+  const {
+    getDefaultScreenshotpath,
+    getScreenShotSavePath,
+  } = userPreferenceSettings;
+  const defaultPath = getDefaultScreenshotpath();
+  const path = getScreenShotSavePath();
+  if (!sortedWebViews.length) return;
+
+  // remove event mirroring between web views
+  pubsub.publish(TOGGLE_EVENT_MIRRORING_ALL_DEVICES, [{status: false}]);
+  // set loader in each webview
+  pubsub.publish(SCREENSHOT_IN_PROGRESS, [{isInProgress: true}]);
+  // store image objects
+  const images = [];
+  // device names
+  const deviceNames = [];
+
+  // get web address of Web views
+  const address = sortedWebViews[0].getURL();
+
+  const fsUtils = new FileSystemUtils(path, defaultPath, now, address);
+
+  for (let i = 0; i < sortedWebViews.length; i += 1) {
```

Comment:
@durgakiran Processing the screenshots sequentially is going to take a very long time. We should look at parallelizing the capture here.

Reply:
As far as I observed, we can't use node workers in electron. I think parallelization in Electron can only be achieved through web workers but web workers don't have access to webviews. So I think we kind of hit a big roadblock there. Do you have any suggestions?

Reply:
I could not find any alternative here

Reply:
@durgakiran Why don't you make it an `async.map` or something like that? 
So that the capture can happen in parallel whenever the CPU is free. We have some intentional `sleeps` in the capture workflow and this parallelization will cut down some waiting time over that.